src/audio/dai.c

```diff
@@ -679,8 +682,9 @@ static int dai_copy(struct comp_dev *dev)
 	} else {
 		src_samples = avail_bytes / get_sample_bytes(dd->frame_fmt);
 		sink_samples = audio_stream_get_free_samples(&buf->stream);
-		copy_bytes = MIN(src_samples, sink_samples) *
-			     get_sample_bytes(dd->frame_fmt);
+		copy_bytes = MIN(dd->period_bytes,
```

Comment:
Starting to get more complex so we will need an inline comment here.

Reply:
Done.